### PR TITLE
Added check for secure connection in all methods.

### PR DESCRIPTION
**Pull request by tizianasellitto**
Every REST API method that needs a jwt token is now checked against
secure connection in the require_token decorator. Every other REST API
method has a require_https decorator. For other request the control is
inside the method.

### CODE REVIEW

File: emotly/utils.py [hunk 1]
```diff
@@ -173,6 +175,16 @@ def check_api_key(*args, **kwargs):
     return check_api_key
 
 
+# Decorator used to check if request is https.
+def require_https(api_method):
+    @wraps(api_method)
+    def check_api_key(*args, **kwargs):
+        if not request.is_secure:
```
Comment (ggc87) [inline emotly/utils.py]: What if we change this condition checking if it is in debug mode? We could close the issue #118618027

File: emotly/controllers/user_controller.py [hunk 2]
```diff
@@ -57,6 +58,12 @@ def login():
 
 @user_controller.route("/signup", methods=["GET", "POST"])
 def signup():
+    url = urlparse(request.url)
+    if not (app.config['DEBUG'] and (url.hostname == "localhost" or
```
Comment (michelangelo) [inline emotly/controllers/user_controller.py]: Do you think it'd be possible to have something more readable here?

File: emotly/controllers/user_controller.py [hunk 3]
```diff
@@ -120,6 +128,12 @@ def register_user(req):
 # with the link to the progressive web app.
 @user_controller.route("/confirm_email/<confirmation_token>", methods=['GET'])
 def confirm_email(confirmation_token):
+    url = urlparse(request.url)
+    if not (app.config['DEBUG'] and (url.hostname == "localhost" or
+                                     url.hostname == "127.0.0.1")) and\
```
Comment (michelangelo) [inline emotly/controllers/user_controller.py]: Ditto.

**Comment by michelangelo** [comment]
Just some minor remark (check the comments). Also, please have a warning (or an assert) in the test runner where we make the tests fail if we run them in debug mode due to the likely possibility of false positives.

Thanks!

File: emotly/controllers/user_controller.py (hunk 2 above)
Comment (tizianasellitto) [inline emotly/controllers/user_controller.py]: I don't actually have found a better way.

**Comment by michelangelo** [comment]
LGTM

**Pull request merged**